vaadinfrontend/src/main/java/life/qbic/datamanager/views/projectOverview/ProjectOverviewHandler.java

```diff
@@ -14,102 +15,116 @@
+    registeredProjectOverview.selectCreationModeDialog.blankButton.addClickListener(
+        e -> creationMode = CreationMode.BLANK);
+    registeredProjectOverview.selectCreationModeDialog.fromOfferButton.addClickListener(
+        e -> creationMode = CreationMode.FROM_OFFER);
+  }
+
+  private void configureSelectionModeDialogFooterButtons() {
+    registeredProjectOverview.selectCreationModeDialog.next.addClickListener(e -> {
+      switch (creationMode) {
+        case BLANK -> {
+          UI.getCurrent().navigate(PROJECT_CREATION_URL);
+          registeredProjectOverview.selectCreationModeDialog.close();
+          registeredProjectOverview.selectCreationModeDialog.reset();
+        }
+        case FROM_OFFER -> {
+          registeredProjectOverview.selectCreationModeDialog.close();
+          loadItemsWithService(offerLookupService);
+          registeredProjectOverview.searchDialog.open();
+        }
+      }
+    });
+    registeredProjectOverview.selectCreationModeDialog.cancel.addClickListener(e -> {
+      registeredProjectOverview.selectCreationModeDialog.close();
+      registeredProjectOverview.selectCreationModeDialog.reset();
+      creationMode = CreationMode.NONE;
+    });
+  }
+
+  private void configureSearchDropbox() {
+    configureSearchDialogFooterButtons();
+
+    registeredProjectOverview.searchDialog.ok.addClickListener(e -> {
+      //check if value is selected
+      registeredProjectOverview.searchDialog.searchField.getOptionalValue()
+          .ifPresent(this::navigateToProjectCreation);
+
+    });
+  }
+
+  private void configureSearchDialogFooterButtons() {
+    registeredProjectOverview.searchDialog.cancel.addClickListener(
+        e -> registeredProjectOverview.searchDialog.close());
+
+    registeredProjectOverview.create.addClickListener(
+        e -> registeredProjectOverview.selectCreationModeDialog.open());
+
+  }
+
+  private void navigateToProjectCreation(OfferPreview offerPreview) {
+    registeredProjectOverview.searchDialog.close();
+    QueryParameters queryParameters = QueryParameters.fromString(
+        OFFER_ID_QUERY_PARAMETER + QUERY_PARAMETER_SEPARATOR + offerPreview.offerId().id());
+
+    UI.getCurrent().navigate(PROJECT_CREATION_URL, queryParameters);
+  }
+
+  private void loadItemsWithService(OfferLookupService service) {
+    registeredProjectOverview.searchDialog.searchField.setItems(
+        query -> service.findOfferContainingProjectTitleOrId(query.getFilter().orElse(""),
+            query.getFilter().orElse(""), query.getOffset(), query.getLimit()).stream());
+
+    registeredProjectOverview.searchDialog.searchField.setRenderer(
+        new ComponentRenderer<>(preview ->
+            new Text(preview.offerId().id() + ", " + preview.getProjectTitle().title())));
+
+    registeredProjectOverview.searchDialog.searchField.setItemLabelGenerator(
+        (ItemLabelGenerator<OfferPreview>) preview ->
+            preview.offerId().id() + ", " + preview.getProjectTitle().title());
```

Comment:
I think it would be helpful to add a small doc or encapsulate it in an expressive method why you need to do the same twice here
(I think one is so that the combobox displays the options, and the other so that the value in the combobox is correctly rendered?)